src/crypto/CHIPCryptoPAL.h

```diff
@@ -541,6 +549,54 @@ class P256Keypair : public P256KeypairBase
     bool mInitialized = false;
 };
 
+using Aes128KeyByteArray = uint8_t[CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES];
+
+/**
+ * @brief Platform-specific AES key
+ *
+ * The class represents AES key used by Matter stack either in the form of raw key material or key
+ * reference, depending on the platform. To achieve that, it contains an opaque context that can be
+ * cast to a concrete representation used by the given platform. Note that currently Matter uses
+ * 128-bit symmetric keys only.
+ */
+class Aes128KeyHandle
+{
+public:
+    Aes128KeyHandle() = default;
+    ~Aes128KeyHandle() { ClearSecretData(mContext.mOpaque); }
+
+    Aes128KeyHandle(const Aes128KeyHandle &) = delete;
+    Aes128KeyHandle(Aes128KeyHandle &&)      = delete;
+    void operator=(const Aes128KeyHandle &) = delete;
+    void operator=(Aes128KeyHandle &&) = delete;
+
+    /**
+     * @brief Get internal context cast to the desired key representation
+     */
+    template <class T>
+    const T & As() const
+    {
+        return *SafePointerCast<const T *>(&mContext);
+    }
+
+    /**
+     * @brief Get internal context cast to the desired, mutable key representation
+     */
+    template <class T>
+    T & AsMutable()
+    {
+        return *SafePointerCast<T *>(&mContext);
+    }
+
+private:
+    static constexpr size_t kContextSize = CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES;
+
+    struct alignas(size_t) OpaqueContext
```

Comment:
```suggestion
    struct alignas(uintptr_t) OpaqueContext
```

Reply:
Must be big enough for at least a poinetr, and therefore, needs pointer alignment